weasyl/controllers/moderation.py

```diff
@@ -219,3 +219,192 @@ def modcontrol_copynotetostaffnotes_post_(request):
+    """
+    query = define.engine.execute("""
+        SELECT lo.login_name, su.submitid, su.title, su.content
+        FROM login lo
+        INNER JOIN submission su USING (userid)
+        WHERE su.is_spam
+            AND su.settings !~ 'h'
+    """).fetchall()
+    return Response(define.webpage(
+        request.userid,
+        "modcontrol/spamqueue/submission.html",
+        [query],
+        title="Submission Spam Queue"
+    ))
+
+
+@moderator_only
+@token_checked
+def modcontrol_spamqueue_submission_post_(request):
+    """
+    A Pyramid route controller for approving for display or items which the spam filter has tagged as being spam.
+    :param request: The Pyramid request.
+    :return/raises: HTTPSeeOther for the journal spam queue.
+    """
+    action = request.params.get("directive")
+    submitid = request.params.get("submitid")
+
+    if not all([action, submitid]):
+        raise WeasylError("Unexpected")
+
+    if action == "approve":
+        # Approve and insert the journal into the notifications table.
+        submitid, userid, rating, settings, content, ua_id, ip_addr = define.engine.execute("""
+            UPDATE submission
+            SET is_spam = FALSE
+            WHERE submitid = %(id)s
+            RETURNING submitid, userid, rating, settings, content, submitter_user_agent_id, submitter_ip_address
+        """, id=submitid).first()
+        # Update the spam filtering backend to indicate that this was not spam
+        spam_filtering.submit(
+            is_ham=True,
+            user_ip=ip_addr,
+            user_agent_id=ua_id,
+            user_id=userid,
+            comment_type="submission",
+            comment_content=content,
+        )
+        welcome.submission_insert(userid=userid, submitid=submitid, rating=rating, settings=settings)
+    elif action == "reject":
+        # Note: We need to explicitly convert to int for the moment, otherwise this call fails with:
+        #    TypeError: %d format: a number is required, not str
+        moderation.hidesubmission(submitid=int(submitid))
+    else:
+        raise HTTPSeeOther("/modcontrol/spamqueue/submission")
+
+    raise HTTPSeeOther("/modcontrol/spamqueue/submission")
+
+
+@moderator_only
+@token_checked
+def modcontrol_spam_remove_post_(request):
+    """
+    Submits content to the spam filtering backend, and hides it from view.
+
+    Either `submitid` or `journalid` must be present in the request's parameters.
+
+    :param request: The Pyramid request.
+    :subparam request.params['submitid']: If present, the submission's ID number.
+    :subparam request.params['journalid']: If present, the journal's ID number.
+    :return/raises: HTTPSeeOther to /modcontrol/suspenduser.
+    """
+    submitid = request.params.get('submitid')
+    journalid = request.params.get('journalid')
+
+    # Only one parameter should ever be set
+    if sum(item is not None for item in [submitid, journalid]) != 1:
+        raise WeasylError("Unexpected")
```

Comment:
Similar here – ids being converted to `int` as late as possible is weird.

```suggestion
        raise WeasylError("Unexpected")

    submitid = int(submitid)
    journalid = int(journalid)
```